modules/pricer/src/main/java/com/opengamma/strata/pricer/swaption/VolatilitySwaptionTradePricer.java

```diff
@@ -0,0 +1,166 @@
+      DiscountingPaymentPricer paymentPricer) {
+
+    this.productPricer = ArgChecker.notNull(productPricer, "productPricer");
+    this.paymentPricer = ArgChecker.notNull(paymentPricer, "paymentPricer");
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the present value of the swaption trade.
+   * <p>
+   * The result is expressed using the currency of the swaption.
+   * 
+   * @param trade  the swaption trade
+   * @param ratesProvider  the rates provider
+   * @param swaptionVolatilities  the volatilities
+   * @return the present value of the swap product
+   */
+  public CurrencyAmount presentValue(
+      ResolvedSwaptionTrade trade,
+      RatesProvider ratesProvider,
+      SwaptionVolatilities swaptionVolatilities) {
+
+    ResolvedSwaption product = trade.getProduct();
+    CurrencyAmount pvProduct = productPricer.presentValue(product, ratesProvider, swaptionVolatilities);
+    Payment premium = trade.getPremium();
+    CurrencyAmount pvPremium = paymentPricer.presentValue(premium, ratesProvider);
+    return pvProduct.plus(pvPremium);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Computes the currency exposure of the swaption trade
+   * 
+   * @param trade  the swaption trade
+   * @param ratesProvider  the rates provider
+   * @param swaptionVolatilities  the volatilities
+   * @return the present value of the swaption product
+   */
+  public MultiCurrencyAmount currencyExposure(
+      ResolvedSwaptionTrade trade,
+      RatesProvider ratesProvider,
+      SwaptionVolatilities swaptionVolatilities) {
+
+    return MultiCurrencyAmount.of(presentValue(trade, ratesProvider, swaptionVolatilities));
+  }
+
+  /**
+   * Calculates the current of the swaption trade.
+   * <p>
+   * Only the premium is contributing to the current cash for non-cash settle swaptions.
+   * 
+   * @param trade  the swaption trade
+   * @param valuationDate  the valuation date
+   * @return the current cash amount
+   */
+  public CurrencyAmount currentCash(ResolvedSwaptionTrade trade, LocalDate valuationDate) {
+    Payment premium = trade.getPremium();
+    if (premium.getDate().equals(valuationDate)) {
+      return CurrencyAmount.of(premium.getCurrency(), premium.getAmount());
+    }
+    return CurrencyAmount.of(premium.getCurrency(), 0d);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the present value sensitivity of the swaption trade.
+   * <p>
+   * The present value sensitivity of the product is the sensitivity of the present value to
+   * the underlying curves.
+   * 
+   * @param trade  the swaption trade
+   * @param ratesProvider  the rates provider
+   * @param swaptionVolatilities  the volatilities
+   * @return the present value curve sensitivity of the swap trade
+   */
+  public PointSensitivityBuilder presentValueSensitivityStickyStrike(
```

Comment:
Is it documented anywhere what the expected unit for a call like this is? As a user I would find it useful for the method to state what the magnitude of the sensitivity is, e.g. to 1bp, to 1%, 100% etc. If not stated, I would probably assume 1bp, though I know this isn't always the case at the pricer level...